### PR TITLE
Build polish round 2

### PR DESCRIPTION
**Pull request by bennybp**
Work in progress. Feel free to comment and add TODOs as necessary

Some polishing of the build system

Done so far
- More printing of what was found and where
- Move pulsar modules, etc, to lib/pulsar
- Created meta project
- Thorough cleaning

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,43 +32,40 @@ endif()
...
 
 
-#############
-# PYTHON
-#############
-set(Python_ADDITIONAL_VERSIONS "3.3;3.4;3.5;3.6")
+#####################
+# Python libraries
+#####################
+set(Python_ADDITIONAL_VERSIONS "3.3;3.4;3.5;3.6;3.7")
```
Comment (loriab) [inline CMakeLists.txt]: pybind11, at least, investigated and decided reverse order was right

**Comment by bennybp** [comment]
Ok this is basically ready to go I believe. Take the new meta project for a spin and let me know if there's any problems or improvements.

I created a little build documentation in the meta project, but everything should be straightforward.

**Comment by ryanmrichard** [comment]
Ok.  Will do (probably tomorrow morning, just landed in ATL).

Sent from my Android (without spellcheck).  Please forgive the brevity.

On Jan 17, 2017 2:09 PM, "Ben Pritchard" <notifications@github.com> wrote:

Ok this is basically ready to go I believe. Take the new meta project for a
spin and let me know if there's any problems or improvements.

I created a little build documentation in the meta project, but everything
should be straightforward.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/pulsar-chem/Pulsar-Core/pull/29#issuecomment-273267057>,
or mute the thread

**Comment by ryanmrichard** [comment]
Sorry for the delay, I just tried meta and it seems to work fine, but then again all of my stuff is installed in the default places.  Did you intend meta to ensure all dependencies of Pulsar are found up front?  If so, you need Berkeley DB, Eigen, and TBB too.  For some reason I seem to vaguely recall discussing this with you at some point, but I can't remember the answer.

**Comment by bennybp** [comment]
I've been somewhat leaning towards not including it in the superbuild, as it would then be duplicated in several places (ie, in the meta build, the core superbuild, and then finally the core project). The only downside would be not being able to get errors right up front.

**Pull request merged**